app.rb

```diff
@@ -0,0 +1,143 @@
+require './book'
+require './student'
+require './teacher'
+require './person'
+require './rental'
+
+class App
+  include Inputs
+  def initialize
+    @people = []
+    @books = []
+    @id = 0
+  end
+
+  attr_accessor :persons, :books
+
+  def run
+    puts 'Welcome to School Library App!'
+    display_options
+
+    task_list = {
+      '1' => method(:list_all_books),
+      '2' => method(:list_all_people),
+      '3' => method(:create_person),
+      '4' => method(:create_book),
+      '5' => method(:create_rental),
+      '6' => method(:list_rentals)
+    }
+
+    task = gets.chomp
+    if task.to_i.positive? && task.to_i < 7
+      task_list[task].call
+    elsif task == '7'
+      puts 'Thank you for using this app!'
+    else
+      puts 'Please select a valid number'
+      run
+    end
+  end
+
+  def display_options
+    puts 'Please choose an option:'
+    puts '1 - List all books'
+    puts '2 - List all people'
+    puts '3 - Create a person'
+    puts '4 - Create a book'
+    puts '5 - Create a rental'
+    puts '6 - List all rentals for a given person id'
+    puts '7 - Exit'
+  end
+
+  def list_all_books
+    puts "\n"
+    books.each_with_index do |book, index|
+      puts "[#{index + 1}] Title: #{book.title}, Author: #{book.author}"
+    end
+  end
+
+  def list_all_people
+    @people.each_with_index do |person, index|
+      puts "#{index}) [#{person.class.name}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+    end
+  end
+
+  def create_person
+    print 'Do you want to create a student (1) or a teacher (2)? [Input the number]: '
+    person_selected = gets.chomp.strip.to_i
+
+    case person_selected
+    when 1
+      create_student
+    when 2
+      create_teacher
+    else
+      puts 'Invalid Selection. Returning to main menu'
+      nil
+    end
+    puts "Person created successfully\n\n"
+  end
+
+  def create_student
+    age = numeric(message: "Age:\s")
+    name = not_empty(message: "Name:\s")
```

Comment:
- [ ] There are no ruby inbuilt methods called `numeric` or `not_empty`. The `not_empty?` method is a method that can be called on an array to check if it is not empty. I also checked all the ruby files in your directory and there are no custom functions called `numeric` and `not_empty`. Kindly recheck the methods you are calling here. 

#### Kindly apply this change for all occurrences where these methods were called. 👍